src/main/java/com/google/devtools/build/lib/analysis/constraints/TopLevelConstraintSemantics.java

```diff
@@ -60,6 +62,21 @@
   private final Function<Key, BuildConfiguration> configurationProvider;
   private final ExtendedEventHandler eventHandler;
 
+  /** Targets that have additional restrictions based on the current platform. */
+  public class PlatformRestrictionsResult {
```

Comment:
Nit: Use AutoValue, make this immutable. Use a Builder if needed to accumulate the values.

Reply:
Done